### PR TITLE
Fix Python 2 to 3 compatibility issues in pokedex codebase. 

### PR DESCRIPTION
**Pull request by LautiLosio**
Specificaly, replace the use of the `next()` method with `__next__()` and ensure that all strings are of type `str`. Also, update the `download_database()` function to use the `gzip` module instead of `zlib`.

### CODE REVIEW

**Comment by LautiLosio** [comment]
Hope this gets through. This is my firs contribution!

File: pokedex/database/get.py [hunk 1]
```diff
@@ -3,11 +3,37 @@
 import os
 import requests
 import zlib
```
Comment (Tenchi2xh) [inline pokedex/database/get.py]: `zlib` is now unused

File: pokedex/database/get.py [hunk 2]
```diff
@@ -3,11 +3,37 @@
...
+#             if chunk:
+#                 gzipped += chunk
+#                 bytes_done += len(chunk)
+#                 bar.update(bytes_done)
+
+#     decompressed_data = zlib.decompress(gzipped, 16+zlib.MAX_WBITS)
+
+#     with open(target, "wb") as file:
+#         file.write(decompressed_data)
+
```
Comment (Tenchi2xh) [inline pokedex/database/get.py]: No need to keep this

**Review by Tenchi2xh — CHANGES_REQUESTED** [review]
Nice, thanks. I should rewrite this whole project to more modern standards one day :D

File: pokedex/database/get.py (hunk 2 above)
Comment (LautiLosio) [reply pokedex/database/get.py]: Ah yes, i should have removed it!

**Pull request merged**